### PR TITLE
Fabo/4873 query rewards from our API not polkadot to stitch claim txs

### LINKED ISSUE #4873: Use DB rewards for claim tx rewards to improve performance

### PR DESCRIPTION
**Pull request by faboweb**
Closes #4873

Needs #4874 

**Description:**

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

**Comment by faboweb** [comment]
🚀 Deployed on https://5f5a70b77d272500b16216bc--lunieio.netlify.app

**Comment by faboweb** [comment]
missing extension logic

File: api/lib/source/polkadotV0-source.js [hunk 1]
```diff
@@ -333,37 +332,36 @@ class polkadotAPI {
...
     const filteredRewards = rewards.filter(({ height: era, validator }) => {
-      if (Number(era) <= MIGRATION_HEIGHT) {
-        return !userClaimedRewards.includes(Number(era))
-      }
       return (
         !stakingLedgers[validator] ||
         !stakingLedgers[validator].includes(Number(era))
       )
     })
+    // console.log(rewards, filteredRewards)
```
Comment (faboweb) [inline api/lib/source/polkadotV0-source.js:354]:
```suggestion
```

File: api/lib/source/polkadotV0-source.js [hunk 2]
```diff
@@ -333,37 +332,36 @@ class polkadotAPI {
       allStakingLedgers[stashId] = result.stakingLedger.claimedRewards
     }
 
+    // console.log(JSON.stringify(allStakingLedgers, null, 2))
```
Comment (faboweb) [inline api/lib/source/polkadotV0-source.js:335]:
```suggestion
```

File: api/data/network-fees.js [hunk 3]
```diff
@@ -216,7 +217,7 @@ const getPolkadotMessage = async (messageType, senderAddress, message, network,
   const messageFormatter = polkadotMessages[messageType]
   const api = networkSource.store.polkadotRPC
   await api.isReady
-  return messageFormatter && network ? await messageFormatter(senderAddress, api, message, network) : null
+  return messageFormatter && network ? await messageFormatter(senderAddress, api, message, network, networkSource) : null
```
Comment (faboweb) [inline api/data/network-fees.js:220]:
```suggestion
  return messageFormatter && network ? await messageFormatter(senderAddress, api, message, network) : null
```

File: api/lib/source/polkadotV0-source.js [hunk 4]
```diff
@@ -333,37 +332,36 @@ class polkadotAPI {
...
   }
 
-  async filterRewards(rewards, delegatorAddress) {
-    const api = await this.getAPI()
+  async filterRewards(rewards) {
     if (rewards.length === 0) {
       return []
     }
     const allValidators = rewards.map(({ validator }) => validator)
-    // DEPRECATE at era 718 + 84
```
Comment (faboweb) [inline api/lib/source/polkadotV0-source.js:345]: we missed to deprecate this ad it affects mainnet

**Comment by faboweb** [comment]
the api changes are only linting

**Comment by faboweb** [comment]
ready I guess

File: api/data/network-fees.js (hunk 3 above)
Comment (Bitcoinera) [reply api/data/network-fees.js]: :eyes:

File: app/src/ActionModal/components/ActionModal.vue [hunk 5]
```diff
@@ -696,7 +696,7 @@ export default {
           password: this.password,
           polkadotAPI,
           HDPath,
-          curve,
+          curve
```
Comment (Bitcoinera) [inline app/src/ActionModal/components/ActionModal.vue:699]: :facepalm:  linting inconsistencies again

File: app/src/signing/transaction-manager.js [hunk 6]
```diff
@@ -204,25 +204,6 @@ export default class TransactionManager {
...
-
-    const { partialFee } = await chainMessage.transaction.paymentInfo(
-      senderAddress
-    )
-    const chainFees = partialFee.toJSON()
-    const viewFees = BigNumber(chainFees)
-      .times(network.coinLookup[0].chainToViewConversionFactor)
-      .toNumber()
-    return viewFees
-  }
```
Comment (Bitcoinera) [inline app/src/signing/transaction-manager.js:225]: What was this doing here? Totally deprecated code

**Comment by Bitcoinera** [comment]
It gets hung up on the E2E deploy for some reason

**Review by Bitcoinera — APPROVED** [review]
Tested it and works smoothly

**Pull request merged**